crypto/src/hash/rescue_prime/mod.rs

```diff
@@ -0,0 +1,816 @@
+use crate::alloc::vec::Vec;
```

Comment:
It's better not to have a giant mod.rs file.
Instead, it's better to have a separate file with each struct or enum.

Reply:
+1

Reply:
Restructured the files as suggested. Now is more maintainable and clear